src/index.js

```diff
@@ -517,76 +536,206 @@ class BouyguesTelecomContentScript extends ContentScript {
     }
   }
 
-  async fetchIdentity(isActive) {
-    this.log('info', 'fetchIdentity starts')
-    let mailAddress
-    let phoneNumber
-    const infosElements = Array.from(
-      document.querySelectorAll(
-        '.personalInfosAccountDetails .tiles .segment:not(.flexCenter)'
+  async checkSessionStorageForIdentity() {
+    this.log('info', '📍️ checkSessionStorageForIdentity starts')
+    let wantedKeys = {}
+    let counter = 0
+    let shouldExit = false
+
+    // Tester le sessionStorage toutes les secondes pendant 5 secondes
+    await new Promise(resolve => {
```

Comment:
Why don't you use wait-for here? I think it would make the code much more readable.